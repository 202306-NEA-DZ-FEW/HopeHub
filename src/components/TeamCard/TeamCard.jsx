import Image from "next/image";
import React from "react";

const TeamCard = ({ image, name, position }) => {
    return (
        <div className='card w-48  shadow-xl mb-4 m-4 bg-Accent hover:bg-Primary hover:shadow-md'>
            <figure>
                <Image
                    src={image}
                    alt={name}
                    width='96'
                    height='42'
                    layout='responsive'
                />
            </figure>
            <div className='card-body text-NeutralBlack'>
                <h2 className='card-title text-l'>{name}</h2>
                <p>{position}</p>
            </div>
        </div>
    );
};

export default TeamCard;
