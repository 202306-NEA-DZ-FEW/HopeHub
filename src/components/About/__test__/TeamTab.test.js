import renderer from "react-test-renderer";
import Abdelghani from "../../../../public/assets/Team/Abdelghani.jpg";

import TeamCard from "../TeamTab";

// Sample team members data for testing
const mockTeamMembers = [
    {
        name: "John Doe",
        image: Abdelghani,
        linkedin: "https://www.linkedin.com/in/johndoe/",
        github: "https://github.com/johndoe",
    },
    {
        name: "Jane Smith",
        image: Abdelghani,
        linkedin: "https://www.linkedin.com/in/janesmith/",
        github: "https://github.com/janesmith",
    },
    // Add more members as needed
];

// Your test case using the mock data
it("renders correctly", () => {
    const tree = renderer
        .create(<TeamCard teamMembers={mockTeamMembers} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
