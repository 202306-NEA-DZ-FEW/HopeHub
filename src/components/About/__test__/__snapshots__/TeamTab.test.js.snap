// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-center"
>
  <div
    className="w-52 h-auto rounded-lg p-2 flex flex-col items-center justify-center mx-2 hover:scale-110"
    style={
      Object {
        "background": "linear-gradient(to top, #BFDFDC, #99B4DF)",
      }
    }
  >
    <img
      alt="John Doe"
      className="w-36 h-36 rounded-lg object-cover my-4"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={500}
    />
    <h2
      className="text-sm font-poppins mt-4 text-NeutralBlack dark:text-NeutralWhite"
    >
      John Doe
    </h2>
    <div
      className="mt-2 flex flex-row py-3"
    >
      <a
        className="text-blue-500 hover:text-blue-700 mr-4"
        href="https://www.linkedin.com/in/johndoe/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          className="text-2xl hover:scale-105 delay-75"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.146C0 .513.526 0 1.175 0h13.65C15.474 0 16 .513 16 1.146v13.708c0 .633-.526 1.146-1.175 1.146H1.175C.526 16 0 15.487 0 14.854V1.146zm4.943 12.248V6.169H2.542v7.225h2.401zm-1.2-8.212c.837 0 1.358-.554 1.358-1.248-.015-.709-.52-1.248-1.342-1.248-.822 0-1.359.54-1.359 1.248 0 .694.521 1.248 1.327 1.248h.016zm4.908 8.212V9.359c0-.216.016-.432.08-.586.173-.431.568-.878 1.232-.878.869 0 1.216.662 1.216 1.634v3.865h2.401V9.25c0-2.22-1.184-3.252-2.764-3.252-1.274 0-1.845.7-2.165 1.193v.025h-.016a5.54 5.54 0 0 1 .016-.025V6.169h-2.4c.03.678 0 7.225 0 7.225h2.4z"
          />
        </svg>
      </a>
      <a
        className="text-gray-700 dark:text-gray-300 hover:text-gray-900 dark:hover:text-gray-100"
        href="https://github.com/johndoe"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          className="text-2xl hover:scale-125 delay-75"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="w-52 h-auto rounded-lg p-2 flex flex-col items-center justify-center mx-2 hover:scale-110"
    style={
      Object {
        "background": "linear-gradient(to bottom, #BFDFDC, #99B4DF)",
      }
    }
  >
    <img
      alt="Jane Smith"
      className="w-36 h-36 rounded-lg object-cover my-4"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={500}
    />
    <h2
      className="text-sm font-poppins mt-4 text-NeutralBlack dark:text-NeutralWhite"
    >
      Jane Smith
    </h2>
    <div
      className="mt-2 flex flex-row py-3"
    >
      <a
        className="text-blue-500 hover:text-blue-700 mr-4"
        href="https://www.linkedin.com/in/janesmith/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          className="text-2xl hover:scale-105 delay-75"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.146C0 .513.526 0 1.175 0h13.65C15.474 0 16 .513 16 1.146v13.708c0 .633-.526 1.146-1.175 1.146H1.175C.526 16 0 15.487 0 14.854V1.146zm4.943 12.248V6.169H2.542v7.225h2.401zm-1.2-8.212c.837 0 1.358-.554 1.358-1.248-.015-.709-.52-1.248-1.342-1.248-.822 0-1.359.54-1.359 1.248 0 .694.521 1.248 1.327 1.248h.016zm4.908 8.212V9.359c0-.216.016-.432.08-.586.173-.431.568-.878 1.232-.878.869 0 1.216.662 1.216 1.634v3.865h2.401V9.25c0-2.22-1.184-3.252-2.764-3.252-1.274 0-1.845.7-2.165 1.193v.025h-.016a5.54 5.54 0 0 1 .016-.025V6.169h-2.4c.03.678 0 7.225 0 7.225h2.4z"
          />
        </svg>
      </a>
      <a
        className="text-gray-700 dark:text-gray-300 hover:text-gray-900 dark:hover:text-gray-100"
        href="https://github.com/janesmith"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          className="text-2xl hover:scale-125 delay-75"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
